openedx_external_enrollments/tests/api/v0/test_views.py

```diff
@@ -0,0 +1,689 @@
+        self.assertRaises(NotImplementedError, self.base._get_enrollment_headers)
+
+    def test_get_enrollment_url(self):
+        """Testing _get_enrollment_url method."""
+        self.assertRaises(NotImplementedError, self.base._get_enrollment_url)
+
+
+class EdxEnterpriseExternalEnrollmentTest(TestCase):
+    """Test class for EdxEnterpriseExternalEnrollment class."""
+
+    def setUp(self):
+        """Set test database."""
+        self.base = EdxEnterpriseExternalEnrollment()
+
+    def test_get_enrollment_data(self):
+        """Testing _get_enrollment_data method."""
+        data = {
+            'course_mode': 'first_mode',
+            'user_email': 'test_email',
+        }
+        course_settings = {
+            'external_course_run_id': 'test_course_run_id',
+            'external_enrollment_mode_override': 'test_mode',
+        }
+        expected_data = {
+            'course_run_id': 'test_course_run_id',
+            'course_mode': 'test_mode',
+            'user_email': 'test_email',
+            'is_active': True,
+        }
+
+        self.assertEqual(
+            self.base._get_enrollment_data(data, course_settings),
+            [expected_data],
+        )
+
+        del course_settings['external_enrollment_mode_override']
+        expected_data['course_mode'] = data['course_mode']
+
+        self.assertEqual(
+            self.base._get_enrollment_data(data, course_settings),
+            [expected_data],
+        )
+
+    @patch.object(EdxEnterpriseExternalEnrollment, '_execute_post')
+    def test_get_enrollment_headers(self, post_mock):
+        """Testing _get_enrollment_headers method."""
+        data = OrderedDict(
+            grant_type="client_credentials",
+            client_id=settings.EDX_ENTERPRISE_API_CLIENT_ID,
+            client_secret=settings.EDX_ENTERPRISE_API_CLIENT_SECRET,
+            token_type="jwt",
+        )
+        post_mock.return_value.json.return_value = {
+            'token_type': 'test-token-type',
+            'access_token': 'test-access-token',
+        }
+        expected_headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'Authorization': 'test-token-type test-access-token',
+        }
+
+        self.assertEqual(self.base._get_enrollment_headers(), expected_headers)
+        post_mock.assert_called_with(settings.EDX_ENTERPRISE_API_TOKEN_URL, data)
+
+        post_mock.return_value.ok = False
+        self.assertIsNone(self.base._get_enrollment_headers())
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            post_mock.side_effect = Exception('test-exception')
+            self.assertIsNone(self.base._get_enrollment_headers())
+            log_capture.check(
+                ('openedx_external_enrollments.api.v0.views', 'ERROR', 'Failed to get token: test-exception'),
+            )
+
+    def test_get_enrollment_url(self):
+        """Testing _get_enrollment_url method."""
```

Comment:
when what ? if we don't have settings ? if the value doesn't exist this raises the exception for that

Reply:
Yes sorry, when you don't have the settings. Should we test that it raises that exception? @andrey-canon

Reply:
no, we shouldn't since the method doesn't handle exceptions

Reply:
the method doesn't raise any exceptions and we are testing the method not the case if the settings are wrong, this could be done as a different test case for settings.

Reply:
Could you write the test case for wrong settings? @andrey-canon

Reply:
it doesn't make sense since the method doesn't handle exceptions

Reply:
Ok.